### DISCUSSION TITLE
Event listener not working

### DISCUSSION THREAD

**Discussion by giacomomasseron** [livewire/livewire#8066] [1. Help] [answered]
Hi,  
in the same page I have 2 components.  
I need to dispatch event from the first one to the second one.

This is the first:  

    info('event dispatched');
    $this->dispatch('cart-updated')->to(TopBar::class);

I am sure this code is executed, cause I see the log.

This is the second one:

```
    #[On('cart-updated')]
    public function cartUpdated(): void
    {
        info('Event listened');
    }
```
The second log is never written.

**Comment by danie-ramdhani** [comment, marked as answer]
made a clone for you case: https://wirebox.app/b/40lg0

maybe you forget to import `use Livewire\Attributes\On;`?

**Reply by giacomomasseron** [reply]
That's it, thank you!
